### ISSUE TITLE
Static library deprecated?

### ISSUE THREAD

**Issue by zc62** [bast/numgrid#28]
I found that in d0edb394e6d1e2ca6c322d3fd903795bd1d1aca9 the static library feature is dropped. I wonder if this is desired, as I think linking to a static library is necessary to produce a executable to share with others.

**Comment by bast** [comment]
Yes, this was probably a mistake :-) - I was thinking too Python-centric. I will think about it a bit more but probably reintroduce it today (now working on preparing `v1.0.0`). Sorry for breaking your code.

**Comment by bast** [comment]
Just out of interest, in which project and how do you use the numgrid code?

**Comment by zc62** [comment]
I write a small piece of C code to extract information such as radial density and others from wfx files that are produced by Gaussian. I have the source code of a larger project that implements the DFT numerical grid itself, but I find it is easier for me to link to a library for this project. I would probably open source my little project soon, in next a few months.

**Comment by bast** [comment]
Thanks! I am waiting for GitHub to be back on track: https://status.github.com/messages and then I will fix these issues.

**Comment by bast** [comment]
This is now hopefully solved - I would appreciate if you could check whether it looks good from your perspective, too.

**Comment by zc62** [comment]
@bast Hi, I just tried this out. It seems that only the shared library is installed, but not the static library.

I am actually using it in a cmake project, I wrote something like
```cmake
include(ExternalProject)
set(EXTERNAL_INSTALL_LOCATION ${CMAKE_BINARY_DIR}/external)

ExternalProject_Add(libnumgrid
    GIT_REPOSITORY https://github.com/dftlibs/numgrid
    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION} -DENABLE_FC_SUPPORT=OFF -DCMAKE_CXX_COMPILER=icpc -DEXTRA_CXXFLAGS="" -DENABLE_CODE_COVERAGE=False -DCMAKE_BUILD_TYPE=release
)

link_directories(${EXTERNAL_INSTALL_LOCATION}/numgrid/lib)
add_executable(xxx xxx.c)
add_dependencies(xxx libnumgrid)
target_include_directories(xxx PUBLIC ${EXTERNAL_INSTALL_LOCATION}/numgrid/include)
target_link_libraries(xxx -static numgrid stdc++ m)
```
During the make procedure, `ld` complains
```
ld: cannot find -lnumgrid
```
In `external/numgrid/lib`, I can only find `libnumgrid.so`. Because I specified `-static`, `ld` is looking for `libnumgrid.a`
The installation log indicates that `libnumgrid.a` is not installed
```
-- Installing: /home/zc62/xxx/build/external/numgrid/lib/libnumgrid.so
-- Installing: /home/zc62/xxx/build/external/lib/libgmock.a
-- Installing: /home/zc62/xxx/build/external/lib/libgmock_main.a
-- Installing: /home/zc62/xxx/build/external/lib/libgtest.a
-- Installing: /home/zc62/xxx/build/external/lib/libgtest_main.a
```
If I manually copy `libnumgrid.a` from `./libnumgrid-prefix/src/libnumgrid-build/lib/libnumgrid.a`, then the compilation is okay.

So is this a bug?

Also I feel that it would be great if `numgrid.h` and `libnumgrid.a` can be installed to `$PREFIX/include` and `$PREFIX/lib`, instead of `$PREFIX/numgrid/*`. For the header file, it can be `$PREFIX/include/numgrid`, so we would write `#include "numgrid/numgrid.h"`. Otherwise we need to provide the compiler explicit `-I` and `-L` arguments. (Anyway, I have to specify them in `CMakeLists.txt`)

The more general question would be, how to properly include numgrid in a cmake project?

**Comment by bast** [comment]
Thanks for feedback and very good questions! I am reopening the issue. This is likely something I need to fix. I will also document how to include numgrid in a CMake project.

**Comment by bast** [comment]
I will try to fix this before the weekend.

**Comment by bast** [comment]
Sorry - it's not forgotten, high on my todo list.

**Comment by bast** [comment]
Documentation of how to include this lib in another CMake project moved to #30. Now finally working on this issue.

**Comment by bast** [comment]
This is fixed in 457dfb3417ef009c36a9627c863f6eb4d6c5043b, both the missing static lib as well as the redundant "numgrid" appended to the prefix. Documentation now also shows how to include the library in a CMake project.

Sorry that it took so long before I started - it was a crazy month.
